src/childrenHavePropXorChildren.js

```diff
@@ -7,22 +7,27 @@ export default function childrenHavePropXorChildren(prop) {
   }
 
   const validator = function childrenHavePropXorChildrenWithProp({ children }, _, componentName) {
-    const childrenCount = React.Children.count(children);
+    let nonFalseyChildrenCount = 0;
```

Comment:
Sorry to nit, but "falsy" is generally spelled without an `e` :-) would you mind changing it throughout?